fix: --version output

bcdady opened the pull request:
retrieves version from plugin.yaml

Closes #335

bcdady committed:
fix: --version output

retrieves version from plugin.yaml

nikpivkin commented on pkg/commands/app.go:
```diff
@@ -122,9 +124,30 @@ The following services are supported:
 	awsFlags.AddFlags(cmd)
 
+	cmd.SetVersionTemplate("{{.Version}}\n")
+	cmd.Flags().BoolP("version", "v", false, "version for aws")
```
Trivy already defines the version flag in the global flag group https://github.com/aquasecurity/trivy/blob/main/pkg/flag/global_flags.go#L22-L28

nikpivkin replied:
What do you think? I think the repeated declaration is redundant.

nikpivkin commented on pkg/commands/app.go:
```diff
@@ -122,9 +124,30 @@ The following services are supported:
+	Version string `yaml:"version"`
+}
+
+func getPluginVersion() string {
```
Maybe it's better to set the version via ldflags, which will be taken from the tag? The tag is the source of the current version.

```makefile
VERSION := $(shell git describe --tags --always --dirty)
VERSION_NO_V := $(shell echo $(VERSION) | sed 's/^v//')

%/trivy-aws:
  ...
  go build -ldflags "-s -w -X 'github.com/aquasecurity/trivy-aws/internal/version.ver=$(VERSION_NO_V)'" \
  ...
```

`internal/version/version.go`
```go
package version

var ver = "dev"

func Version() string {
	return ver
}
```

nikpivkin commented on pkg/commands/app.go:
```diff
@@ -122,9 +124,30 @@ The following services are supported:
+		return "unknown"
+	}
+	
+	return fmt.Sprintf("Version: %s", config.Version)
```
Formatting should be done in a template.

bcdady replied:
I'm not sure what you mean. The Makefile was modified in a different PR.

bcdady committed:
refactor: use ldflags for version injection instead of runtime file reading

- Add internal/version package with build-time version injection
- Update Makefile to inject version via ldflags from git tags
- Remove runtime plugin.yaml reading for version information
- Use cobra version template for consistent formatting

bcdady committed:
style: gofmt, go mod tidy

bcdady committed:
chore: update Makefile to derive VERSION from git

bcdady committed:
fix(flags): remove redundant version flag from app.go

bcdady committed:
Revert changes from commit e53d40bfdd9bb07fd78db6be3ef8843e789de369 except for internal/version/version.go

nikpivkin committed:
Merge branch 'main' into fix/version-flag
